components/program/Timeline.vue

```diff
@@ -0,0 +1,355 @@
+<script setup lang="ts">
+  import type { AlertType } from 'composables/useAlerts'
+
+  const useAuth = useAuthStore()
+
+  const companies = await $fetch('/api/company', { method: 'GET' })
+
+  const events = ref({})
+  const eventsByDate: Ref<{ [key: string]: any[] }> = ref({})
+  const dates = ref([])
+  const fetchAndUpdateEvents = async () => {
+    try {
+      const updatedEvents = await getEvents()
```

Comment:
Same here. Instead of fetching the data in the onMounted hook you should fetch it at the top level of the setup script using `useFetch` and handle errors without a try catch block. Again, see the documentation.